### DISCUSSION TITLE
Cluster Question

### DISCUSSION THREAD

**Discussion by coolman888** [coinfoundry/miningcore#1119] [Q&A]
Hi, I know MC is using ShareRelay push the shares from slave to master, I would like to know how about the pool hashrate/miner hashrate/ block of the slave pool , Does MC publish these informations to master? or I have to setup the PostgreSQL on slave, remote to master?

**Comment by oliverw** [comment]
Database connectivity is only required on master.

**Reply by ei8ht187** [reply]
Stupid Question: what happens, when the master has to be rebootet or fails for any reason? I haven't found a way to set up an HA Masternode until now.

**Reply by oliverw** [reply]
> Stupid Question: what happens, when the master has to be rebootet or fails for any reason? I haven't found a way to set up an HA Masternode until now.

All ingress data is lost as long as the primary master is down.

**Reply by xmrstickers** [reply]
Is there any documentation/advice on setting up two masters so one can go down without interruption to miners? 
I have been digging through old posts trying to learn as much as possible; please forgive me if I missed it.

**Reply by oliverw** [reply]
> Is there any documentation/advice on setting up two masters so one can go down without interruption to miners? I have been digging through old posts trying to learn as much as possible; please forgive me if I missed it.

High availability configuration is not covered anywhere in the docs. Not even I have a setup for that.
